libcore/test/catch2/geometry/CorrectGeometryTest.cpp

```diff
@@ -0,0 +1,132 @@
+/*
+ * This file is part of JuPedSim.
+ *
+ * JuPedSim is free software: you can redistribute it and/or modify
+ * it under the terms of the GNU Lesser General Public License as published by
+ * the Free Software Foundation, either version 3 of the License, or
+ * any later version.
+ *
+ * JuPedSim is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU Lesser General Public License
+ * along with JuPedSim. If not, see <http://www.gnu.org/licenses/>.
+ *
+ **/
+
+#include "geometry/helper/CorrectGeometry.cpp"
+
+#include "geometry/Building.h"
+#include "geometry/Crossing.h"
+#include "geometry/Line.h"
+#include "geometry/SubRoom.h"
+#include "geometry/Wall.h"
+
+#include <catch2/catch.hpp>
+
+TEST_CASE("geometry/Building/correct", "[geometry][Building][correct]")
+{
+    SECTION("correct") {}
+
+
+    SECTION("SplitWall")
+    {
+        SECTION("No split possible")
+        {
+            NormalSubRoom subRoom;
+            REQUIRE_FALSE(geometry::helper::SplitWall(subRoom, Wall{}));
+
+            Wall bigWall(Point(0, 0), Point(20, 0));
+            REQUIRE_FALSE(geometry::helper::SplitWall(subRoom, bigWall));
+
+            Transition transition;
+            subRoom.AddTransition(&transition);
+            REQUIRE_FALSE(geometry::helper::SplitWall(subRoom, bigWall));
+
+            transition.SetPoint2(Point(0, 10));
+            REQUIRE_FALSE(geometry::helper::SplitWall(subRoom, bigWall));
+
+            Crossing crossing;
+            subRoom.AddCrossing(&crossing);
+            REQUIRE_FALSE(geometry::helper::SplitWall(subRoom, bigWall));
+            crossing.SetPoint1(Point(100, 0));
+            crossing.SetPoint2(Point(100, 1));
+
+            Wall wall;
+            subRoom.AddWall(wall);
+            REQUIRE_FALSE(geometry::helper::SplitWall(subRoom, bigWall));
+
+            wall.SetPoint1(Point(-100, 0));
+            REQUIRE_FALSE(geometry::helper::SplitWall(subRoom, bigWall));
+
+            REQUIRE_FALSE(geometry::helper::SplitWall(subRoom, Wall{}));
+        }
+    }
+
+
+    SECTION("ComputeSplitPoint")
+    {
+        Wall testWall(Point(0, 0), Point(0, 10));
+        SECTION("No Split Point")
+        {
+            // default Walls (0,0)-(0,0)
+            REQUIRE_FALSE(geometry::helper::ComputeSplitPoint(Wall(), Line()));
+            REQUIRE_FALSE(geometry::helper::ComputeSplitPoint(Wall(), Wall()));
+
+            // No intersection
+            REQUIRE_FALSE(geometry::helper::ComputeSplitPoint(testWall, Line()));
+            REQUIRE_FALSE(
+                geometry::helper::ComputeSplitPoint(testWall, Line(Point(1, 1), Point(1, 2))));
+            REQUIRE_FALSE(
+                geometry::helper::ComputeSplitPoint(testWall, Line(Point(0, -1), Point(1, -1))));
+
+            // Overlapping
```

Comment:
Those should really be test sections, if this ever fails due to refactoring you will want this for a much nicer error analysis

Reply:
Done